docs/source/java/memory.rst

```diff
@@ -284,3 +302,159 @@ Finally, enabling the ``TRACE`` logging level will automatically provide this st
+
+Typically, the ReferenceManager implementation used is an instance of `BufferLedger`_. 
+A BufferLedger is a ReferenceManager that also maintains the relationship between an ``AllocationManager``, 
+a ``BufferAllocator`` and one or more individual ``ArrowBuf``\ s
+
+All ArrowBufs (direct or sliced) related to a single BufferLedger/BufferAllocator combination 
+share the same reference count and either all will be valid or all will be invalid. 
+For simplicity of accounting, we treat that memory as being used by one
+of the BufferAllocators associated with the memory. When that allocator
+releases its claim on that memory, the memory ownership is then moved to
+another BufferLedger belonging to the same AllocationManager.
+
+Allocation Details
+------------------
+
+There are several Allocator types in Arrow Java:
+
+-  ``BufferAllocator`` - The public interface application users should be leveraging
+-  ``BaseAllocator`` - The base implementation of memory allocation, contains the meat of the Arrow allocator implementation
+-  ``RootAllocator`` - The root allocator. Typically only one created for a JVM. It serves as the parent/ancestor for child allocators
+-  ``ChildAllocator`` - A child allocator that derives from the root allocator
+
+Many BufferAllocators can reference the same piece of physical memory at the same
+time. It is the AllocationManager’s responsibility to ensure that in this situation, 
+all memory is accurately accounted for from the Root’s perspective
+and also to ensure that the memory is correctly released once all
+BufferAllocators have stopped using that memory.
+
+For simplicity of accounting, we treat that memory as being used by one
+of the BufferAllocators associated with the memory. When that allocator
+releases its claim on that memory, the memory ownership is then moved to
+another BufferLedger belonging to the same AllocationManager. Note that
+because a ArrowBuf.release() is what actually causes memory ownership
+transfer to occur, we always proceed with ownership transfer (even if
+that violates an allocator limit). It is the responsibility of the
+application owning a particular allocator to frequently confirm whether
+the allocator is over its memory limit (BufferAllocator.isOverLimit())
+and if so, attempt to aggressively release memory to ameliorate the
+situation.
+
+
+Object Hierarchy
+----------------
+
+There are two main ways that someone can look at the object hierarchy
+for Arrow’s memory management scheme. The first is a memory based
+perspective as below:
+
+Memory Perspective
+~~~~~~~~~~~~~~~~~~
+
+.. raw:: html
+
+   <pre>
+   + AllocationManager
+   |
+   |-- UnsignedDirectLittleEndian (One per AllocationManager)
+   |
+   |-+ BufferLedger 1 ==> Allocator A (owning)
+   | ` - ArrowBuf 1
+   |-+ BufferLedger 2 ==> Allocator B (non-owning)
+   | ` - ArrowBuf 2
+   |-+ BufferLedger 3 ==> Allocator C (non-owning)
+     | - ArrowBuf 3
+     | - ArrowBuf 4
+     ` - ArrowBuf 5
+   </pre>
```

Comment:
nit: I think you can use `.. code-block:: none` and remove the `<pre>` tags. (Not that I expect anyone to render the docs to anything but HTML…)